### PR TITLE
userbrowse.py: open standard Soulseek path (not URL)

### PR DESCRIPTION
**Pull request by slook**
Soulseek usernames don't seem to allow backslashes in them, so we can use `"\"` as a path separator to navigate to a folder\file within a users shares...

+ Added: Browse to a standard Soulseek path if the entry contains text separated with backslashes `\`
+ ~Fixed: Critical ValueError in browse_queued_path() when browsing to a root share folder, also affects URL browsing in master~

A Soulseek path is assumed to have the syntax `username\share name\folder name\file name`.

Any optional leading `\` before the username are stripped.

If the path ends with a `\` then it is a folder, otherwise it is a file.

If the path only contains a single element, then it is a root folder, regardless if it ends with a `\` or not.

Forward slashes are assumed to be valid characters and are not replaced.

This PR also contains an Alternative fix to #2436

### CODE REVIEW

File: pynicotine/gtkgui/userbrowse.py [hunk 1]
```diff
@@ -473,7 +473,11 @@ def browse_queued_path(self):
         if self.queued_path is None:
             return
 
-        folder, filename = self.queued_path.rsplit("\\", 1)
+        if "\\" in self.queued_path:
+            folder, filename = self.queued_path.rsplit("\\", 1)
+        else:
+            folder, filename = self.queued_path, ""
```
Comment (slook) [inline pynicotine/gtkgui/userbrowse.py:479]: The crash in #2436 can be avoided here like this

File: pynicotine/plugins/core_commands/__init__.py [hunk 2]
```diff
@@ -152,8 +152,8 @@ def __init__(self, *args, **kwargs):
                 "description": _("Browse files of user"),
                 "disable": ["cli"],
                 "group": _CommandGroup.USERS,
-                "usage": ["<user>"],
-                "usage_private_chat": ["[user]"]
+                "usage": ["<user\\folder or url>"],
+                "usage_private_chat": ["[user\\folder]", "[url]"]
```
Comment (mathiascode) [inline pynicotine/plugins/core_commands/__init__.py:156]: Is this syntax based on some existing convention? Not sure I like it, to me it reads like two positional arguments.

**Comment by mathiascode** [comment]
Why have this in addition to URLs? Feels like two solutions competing against each other.

File: pynicotine/plugins/core_commands/__init__.py (hunk 2 above)
Comment (slook) [reply pynicotine/plugins/core_commands/__init__.py]: Agreed, I shouldn't have put it in the Network Filter commands either. It does look confusing.

**Comment by slook** [comment]
> Why have this in addition to URLs?

Because we should use the text entry as an address bar to display the current location, as would be expected of a file manager like interface, and perhaps even implement back/forward history or bookmark favorite location functions, etc.

> two solutions competing against each other

Not really. Both syntaxes are present, so both should be supported. The Windows style paths are more native to the network than the Nicotine+ specific URLs.

A Qt user might need to provide a Windows-like Soulseek path to an N+ user.

**Comment by mathiascode** [comment]
> > Why have this in addition to URLs?
> 
> Both syntaxes are present, so both should be supported. The Windows style paths are more native to the network than the Nicotine+ specific URLs.
> 
> The 'Copy Path' function does not otherwise serve a useful purpose unless it can be pasted here.
> 
> A Qt user might need to provide a Windows-like Soulseek path to an N+ user.

I need to check the history of the 'copy path' menu items, but I'm pretty sure the main purpose was to easily copy the path for use in other programs. Perhaps the file properties dialog would be enough for this nowadays.

There are no paths that start with a username, so you're basically inventing an alternative to slsk URLs. I'd rather not add this.

**Comment by slook** [comment]
All good points. However, I feel like the current current interface is well below par, even if it is not intended to be a fully fledged file manager.

> Is the main issue that you can't see parent folders?

Yes, determining the containing parent folder is difficult. Maybe it would help if the tool tip could say the whole path.

**Comment by mathiascode** [comment]
Closing, as this isn't something I want to add right now.

**Comment by mathiascode** [comment]
> Yes, determining the containing parent folder is difficult. Maybe it would help if the tool tip could say the whole path.

Could be interesting to experiment with a path bar with clickable parts similar to Nautilus, but clicking them would scroll to the desired folder. The path bar could be placed above the folder tree view.

Tooltips showing the full path should be possible now, since we started storing the full path in the treeview model.

**Pull request closed without merging**